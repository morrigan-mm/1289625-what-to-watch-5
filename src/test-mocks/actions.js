export const unknownAction = {type: `NOOP`};
