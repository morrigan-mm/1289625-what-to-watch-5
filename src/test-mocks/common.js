export const mockEvent = {
  preventDefault: () => {}
};

export const noop = () => {};
