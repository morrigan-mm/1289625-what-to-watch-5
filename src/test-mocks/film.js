const film = {
  id: 15,
  poster: `#poster`,
  preview: `#preview`,
  video: `#video`,
  cover: `#cover`,
  title: `The Revenant`,
  genre: `Action`,
  releaseDate: 2015,
  description: `A frontiersman on a fur trading`,
  rating: 4,
  ratingCount: 618498,
  director: `The Director`,
  actors: [`actor1`, `actor2`],
  runtime: 156,
  addedToMyList: false,
  backgroundColor: `#92918B`,
  backgroundImage: `#bgimage`
};

export default film;
