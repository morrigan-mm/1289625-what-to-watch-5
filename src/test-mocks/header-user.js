import {AuthorizationStatus} from "../constants";

export const headerUserAuthorized = {
  authorizationStatus: AuthorizationStatus.AUTH,
  avatar: `#`
};
export const headerUserUnauthorized = {
  authorizationStatus: AuthorizationStatus.NO_AUTH
};
