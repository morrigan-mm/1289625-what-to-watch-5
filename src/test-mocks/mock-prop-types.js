import PropTypes from "prop-types";

export const mockComponentType = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};
