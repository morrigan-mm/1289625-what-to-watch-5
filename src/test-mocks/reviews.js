const reviews = [
  {
    id: 1,
    text: `some review text 1`,
    rating: 1.1,
    author: `Author name1`,
    reviewDate: `2020-01-01T14:13:56.569Z`
  },
  {
    id: 2,
    text: `some review text 2`,
    rating: 2.2,
    author: `Author name2`,
    reviewDate: `2020-02-02T14:13:56.569Z`
  },
  {
    id: 3,
    text: `some review text 3`,
    rating: 3.3,
    author: `Author name3`,
    reviewDate: `2020-03-03T14:13:56.569Z`
  }
];

export default reviews;
