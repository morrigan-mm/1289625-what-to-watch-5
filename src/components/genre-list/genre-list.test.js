import React from "react";
import renderer from "react-test-renderer";
import {noop} from "../../test-mocks/common";
import GenreList from "./genre-list";

const genres = [`Action`, `Crime`];

describe(`GenreList`, () => {
  it(`With no active genre`, () => {
    const tree = renderer
      .create(
          <GenreList genres={genres} onGenreClick={noop} />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`With active genre`, () => {
    const tree = renderer
      .create(
          <GenreList genres={genres} onGenreClick={noop} activeGenre={genres[0]} />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
