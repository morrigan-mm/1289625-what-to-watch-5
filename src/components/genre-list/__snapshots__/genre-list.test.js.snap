// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenreList With active genre 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      data-genre="All Genres"
      href="#"
      onClick={[Function]}
    >
      All Genres
    </a>
  </li>
  <li
    className="catalog__genres-item catalog__genres-item--active"
  >
    <a
      className="catalog__genres-link"
      data-genre="Action"
      href="#"
      onClick={[Function]}
    >
      Action
    </a>
  </li>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      data-genre="Crime"
      href="#"
      onClick={[Function]}
    >
      Crime
    </a>
  </li>
</ul>
`;

exports[`GenreList With no active genre 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item catalog__genres-item--active"
  >
    <a
      className="catalog__genres-link"
      data-genre="All Genres"
      href="#"
      onClick={[Function]}
    >
      All Genres
    </a>
  </li>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      data-genre="Action"
      href="#"
      onClick={[Function]}
    >
      Action
    </a>
  </li>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      data-genre="Crime"
      href="#"
      onClick={[Function]}
    >
      Crime
    </a>
  </li>
</ul>
`;
